src/components/Filters.jsx

```diff
@@ -0,0 +1,62 @@
+import "../styles/filters.css"
+import {useState} from "react"
+import { useFilter } from "../contexts/FilterContext"
+import { useAuth } from "../contexts/AuthContext"
+
+export default function Filters() {
+
+    const {filterState, filterDispatch} = useFilter()
+    const {loggedIn} = useAuth()
+    const [showFilters, setShowFilters] = useState(false)
+ 
+    return (
+        
+        <div className='filter-by'>
+            {loggedIn &&
+            <>
+            <p className="filter-title" onClick={() =>setShowFilters(!showFilters)}>Filter:</p>
+            
+            <div className={showFilters ? 'filter-options shadow active' : 'filter-options shadow'}>
+                <button className="clear-filter" onClick={(e) => filterDispatch({type: "CLEAR"})}>Clear Filters</button>
+                <div className="filter-option">
+                    <p>Labels</p>
+                    <label className="label filter">
+                        <input type="checkbox" name="label" checked={filterState.labels.find(label => label === "Work")}  value="Work" onChange={(e) => filterDispatch({type: "LABEL", payload: e.target.value})}/>Work
+                    </label>
+                    <label className="label filter">
+                        <input type="checkbox" name="label" checked={filterState.labels.find(label => label === "Personal")} value="Personal" onChange={(e) => filterDispatch({type: "LABEL", payload: e.target.value})}/>Personal
+                    </label>
+                    <label className="label filter">
+                        <input type="checkbox" name="label" checked={filterState.labels.find(label => label === "Creativity")} value="Creativity" onChange={(e) => filterDispatch({type: "LABEL", payload: e.target.value})}/>Creativity
+                    </label>
+                    <label className="label filter">
+                        <input type="checkbox" name="label" checked={filterState.labels.find(label => label === "Shopping List")} value="Shopping List" onChange={(e) => filterDispatch({type: "LABEL", payload: e.target.value})}/>Shopping List
+                    </label>
+                    <label className="label filter">
+                        <input type="checkbox" name="label" checked={filterState.labels.find(label => label === "To-do")} value="To-do" onChange={(e) => filterDispatch({type: "LABEL", payload: e.target.value})}/>To-do
+                    </label>
+                    <label className="label filter">
+                        <input type="checkbox" name="label" checked={filterState.labels.find(label => label === "In Progress")} value="In Progress" onChange={(e) => filterDispatch({type: "LABEL", payload: e.target.value})}/>In Progress
+                    </label>
```

Comment:
if it is possible do create your Input component and then use it here you can reduce the code by this.

Reply:
Thank you for pointing this out, totally missed this. I have made the changes.